docs/code/SchemaComprehension.md

```diff
@@ -0,0 +1,210 @@
+| `BL`             | `DvBool`           | `bool`, `bool?`         |
+| `TS`             | `DvTimeSpan`       |                         |
+| `DT`             | `DvDateTime`       |                         |
+| `DZ`             | `DvDateTimeZone`   |                         |
+| Variable-size vector | `VBuffer<T>`   | `T[]`, and the vector is always dense |
+| Fixed-size vector    | `VBuffer<T>` with `[VectorType(N)]` | `T[]` with `VectorType(N)`, and the vector is always dense |
+| Key type             | `uint` with `[KeyType]`             |                                                            |
+
+### Additional attributes to affect type mapping
+
+There are two more attributes that can affect the way ML.NET conducts schema comprehension:
+* `[ColumnName]` lets you choose a different name for the `IDataView` column. By default it is the same as field name.
+  * This is a way to create or read back an `IDataView` column with a name containing 'invalid' characters (like whitespace).
+* `[NoColumn]` is an attribute that denotes that the below field should not be mapped to a column.
+
+### Using SchemaDefinition for run-time type mapping hints
+
+As you can see from the table and notes above, certain `IDataView` types can only be denoted with an additional field attribute. If the type parameters are not known at compile time (like the size of the fixed-size vector), this is tricky. 
+
+You can use a `SchemaDefinition` object to re-map a type to an `IDataView` schema programmatically. It gives you the same powers as the attributes, but at runtime.
+Please see the below example.
+```(csharp)
+// Vector size is only known at runtime.
+int numberOfFeatures = 4;
+
+// Create the default schema definition.
+var schemaDef = SchemaDefinition.Create(typeof(IrisVectorData));
+
+// Specify the right vector size.
+schemaDef["Features"].ColumnType = new VectorType(NumberType.R4, numberOfFeatures);
+
+// Create a data view.
+var dataView = env.CreateDataView<IrisVectorData>(arr, schemaDef);
+
+// Create a prediction engine. You can add custom input and output schema definitions there.
+var predictionEngine = env.CreatePredictionEngine<IrisData, IrisVectorData>(dv, outputSchemaDefinition: schemaDef);
+```
+
+In addition to the above, you can use `SchemaDefinition` to add per-column metadata, or even a 'value generator' (so that the column value is not read from the field, but computed using a delegate).
+
+## Limitations
+
+Certain things are not possible to do at all using the schema comprehensions, but are possible via the native `IDataView` programmatic interface.
+It was our design decision to not allow these scenarios, thus simplifying the other, more common scenarios. 
+
+Here is the list of things that are only possible via the low-level interface:
+* Creating or reading a data view, where even column *types* are not known at compile time (so you cannot create a C# class to define the schema)
```

Comment:
>Creating or reading a data view, where even column *types* are not known at compile time (so you cannot create a C# class to define the schema) [](start = 2, length = 143)

example of scenario when this might occur